### PR TITLE
Wechaty 实现 Milvus 社区群消息同步

### PR DESCRIPTION
**Pull request by shiyu22**
Please make sure to check the following boxes creating your pull request, thank you very much!

- [x] Add Blog Header, including title, author, image, categories, tags, etc.  

    For example:  

    ```md
    ---
    title: Your Great Blog Title
    author: your_github_id
    categories: project
    tags:
      - python
      - padlocal
    image: /assets/2021/02-your-blog-slug/teaser.jpg
    ---

    One line bio

    Your blog post content Your blog post content
    Your blog post content Your blog post content
    Your blog post content Your blog post content
    ```

    > It will be easier to start drafting your post by using a existing post file as your template, for example, [this one](https://github.com/wechaty/wechaty.js.org/blob/master/jekyll/_posts/2021-01-14-wechaty-puppet-service.md)

- [x] Keep all filenames lowercase, and use `-` to connect words instead of space. e.g. `2017-10-06-wechat-pc-impactor.md` instead of `2017-10-06-WeChat PC Impactor.md`
- [ ] Embed video/pdf from YouTube/Bilibili/local (if there's any) by using `{% include iframe.html src="URL_or_PATH" %}`. (Learn more about how to embed file/url in the post by reading this [blog post](https://wechaty.js.org/2020/08/24/add-video-to-wechaty-blog/)
- [x] Save photo to local before publishing, to a folder named by `${MONTH}-${YOUR_BLOG_SLUG}/` under `/assets/2021/` directory.
- [x] Select a beautiful and meaningful teaser image for your blog
- [x] Create your contributor profile (if you are a first time contributor)

    For excample:

    > jekyll/_contributors/lijiarui.md

    ```yaml
    ---
    name: Jiarui LI (李佳芮)
    site: https://lijiarui.github.io
    bio: 句子互动创始人 & CEO，微软人工智能最具价值专家 (AI MVP)
    avatar: /jekyll/assets/contributors/lijiarui/avatar.png
    email: rui@juzi.bot
    twitter: https://weibo.com/u/2175505900
    ---
    ```

  Here is the example file:

  - lijiarui.md: <https://github.com/wechaty/wechaty.js.org/blob/master/jekyll/_contributors/lijiarui.md>
  - It will shown like this on wechaty website: <https://wechaty.js.org/contributors/lijiarui/>

To learn more about the contribute guideline, visit: <https://wechaty.js.org/docs/contributor-program/>

### CODE REVIEW

**Review by huan — CHANGES_REQUESTED** [review]
Thanks for sharing the Wechaty practice experiences from the Milvus community!

Please make the CI green by following the error messages from https://github.com/wechaty/wechaty.js.org/pull/1115/checks?check_run_id=3117089428 before we can continue reviewing this great blog post.

**Review by huan — CHANGES_REQUESTED** [review]
Please convet all PNG files to WEBP format in favor to #1035

**Comment by lijiarui** [comment]
Hi @shiyu22  Please make GitHub action green then we could continue to review this PR.

**Comment by lijiarui** [comment]
Hi, @shiyu22  

The most efficient way to make your post PR turns green is to run the npm test locally before push because it will greatly reduce the time that you spend to pass this CI test.

You can refer:
- Related blog: https://wechaty.js.org/2021/06/23/the-wechaty-way/
- How to run test locally: https://github.com/wechaty/wechaty.js.org#how-to-run-tests

**Comment by lijiarui** [comment]
ping @shiyu22

File: jekyll/_posts/2021-07-20-milvus-sync-bot.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
+---
+title: "Wechaty 实现 Milvus 社区群消息同步"
+
```
Comment (lijiarui) [inline jekyll/_posts/2021-07-20-milvus-sync-bot.md]: Please remove the useless blank here.

File: jekyll/_posts/2021-07-20-milvus-sync-bot.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+author: shiyu22
+
+categories: article
+
+tags:
+
+- blog
+
+- milvus
+
```
Comment (lijiarui) [inline jekyll/_posts/2021-07-20-milvus-sync-bot.md]: Please remove the useless blank here.

File: jekyll/_posts/2021-07-20-milvus-sync-bot.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+> 作者: [shiyu22](https://github.com/shiyu22)， 程序媛
+
+## Wechaty 实现 Milvus 社区群消息同步
+
+## 背景
+
+首先介绍下开发背景，[Milvus](https://milvus.io/) 是一个开源的特征向量数据库，本着开源互助的精神，社区一直致力于帮助用户解疑答惑。企业微信聊天群是社区的交流地之一，基于群聊自动扩群功能下，每个群聊的人数上限是 200 人，所以 Milvus 开源一年半以来，累计了十个用户交流群。用户分散在多个群会发生一些交流或管理方面的问题：
+
+- 旧群交流少：十个交流群中之前的群中交流不多，因为该踩的坑都踩完了，老用户提问的频率不高。
+
```
Comment (lijiarui) [inline jekyll/_posts/2021-07-20-milvus-sync-bot.md]: Please remove the useless blank here. See:   https://docs.google.com/presentation/d/1rWd7aLozgPFyPiSbyxqw1GeGV2-NCGRKvDOUcq9VYcw/edit#slide=id.gec336ad374_1_357

File: jekyll/_posts/2021-07-20-milvus-sync-bot.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+- 技术交流覆盖不广泛：当一个群里交流技术难题，其他群的小伙伴无法参与或查看。
+
+基于以上的这些问题，社区考虑做一个企业微信群间消息同步的机器人，开发群内的消息同步功能，方便群里用户的沟通与交流。
+
+## 效果展示
+
+针对前面提到的问题，社区计划实现一个可以同步群消息的机器人，之前了解到 [Wechaty](https://github.com/wechaty/wechaty) 可以实现这个功能，并且 Wechaty 的社区也部署了同步群消息的机器人。Wechaty 是一个开源的的对话机器人 SDK，它是一个使用 Typescript 构建的Node.js 应用。支持多种微信接入方案，包括网页，ipad，ios，windows， android 等。
+基于 Wechaty 部署的机器人可以同步十个群的消息，效果展示如下面的截图，机器人负责转发其他群的消息，保证每个群可以同步，用户交流群的人就可以跨群交流了。接下来将介绍如何用 Wechaty 部署消息同步的机器人。
+
+<img src="/assets/2021/07-milvus-sync-bot/syncbot1.webp" width = "300" height = "500" alt="图片名称" align=center />
```
Comment (lijiarui) [inline jekyll/_posts/2021-07-20-milvus-sync-bot.md]: Please change to the following:

```
![效果展示](/assets/2021/07-milvus-sync-bot/syncbot1.webp)
```

**Review by lijiarui — CHANGES_REQUESTED** [review]

**Comment by huan** [comment]
Thanks for the update, and I'm looking forward to seeing this blog post online soon!

**Review by huan — CHANGES_REQUESTED** [review]
You are very close to passing the CI!

Please see https://github.com/wechaty/wechaty.js.org/runs/3540785907#step:7:24 and I believe we can see a green CI after fixed all of them.

**Comment by lijiarui** [comment]
ping @shiyu22

**Comment by huan** [comment]
You are very close to passing the CI!

Please see 
 https://github.com/wechaty/wechaty.js.org/pull/1115/checks?check_run_id=3670813600#step:7:24

and I believe we can see a green CI after fixed all of them.

**Comment by huan** [comment]
@shiyu22 It seems that you have only fixed a tiny subset of the linting problems.

I'd like to suggest that you should run the following commands locally (Node.js required):

```sh
npm install
npm test
```

And you will get the linting report before pushing them.

**Comment by lijiarui** [comment]
ping @shiyu22

**Comment by lijiarui** [comment]
Thanks for your update, @shiyu22

In order to make sure your changes have not broken anything and no need to wait for us to click `approve and run`, you can run the following commands before pushing your changes to GitHub.

```shell
npm install
npm test
```

See https://github.com/wechaty/wechaty.js.org#how-to-run-tests

**Review by huan — APPROVED** [review]
LGTM

Congratulations on having fixed all the **linting requirements**!

**Comment by huan** [comment]
Thank you very much for your contribution!

## You are welcome to join [Wechaty Contributor Program](https://wechaty.js.org/docs/contributor-program/)

### 1. Join Wechaty Organization

> You've invited shiyu22 to Wechaty! They'll be receiving an email shortly. They can also visit https://github.com/wechaty to accept the invitation.

I have invited you to join our Wechaty GitHub Organization, please accept it by following the above message. (See also: [wechaty/PMC#16](https://github.com/wechaty/PMC/issues/16))

### 2. Update Your Wechaty Contributor Profile

1. Please open [Contributor Hall of Fame](https://docs.google.com/spreadsheets/d/1XcDoIczyIclqXP1p90Sz7S0n4Q22xjFzJmjPFlU2g1E/edit#gid=0) and add yourself to the end of the list, so that other contributors will know you better!
1. Please add yourself to our Website [Contributors](https://wechaty.js.org/contributors/) by [creating a PR](https://github.com/wechaty/wechaty.js.org/tree/master/jekyll/_contributors) and refer to this PR link as well.

### 3. Join The Contributor Only WeChat Room

We also have a WeChat room for contributors only which can discuss Wechaty at a deeper level, you are welcome to join and if you are interested.

Please add @lijiarui wechat: _ruirui_0914_ and send her this pr link. She will invite you into Wechaty Contributor Room

Cheers!

**Pull request merged**